magefiles/magefile.go

```diff
@@ -622,6 +622,90 @@ func SetupMultiPlatformTests() error {
+		klog.Errorf("failed to get the pipeline bundle ref: %+v", err)
+		return
+	}
+	if tektonObj, err = tekton.ExtractTektonObjectFromBundle(defaultBundleRef, "pipeline", "docker-build"); err != nil {
+		klog.Errorf("failed to extract the Tekton Pipeline from bundle: %+v", err)
+		return
+	}
+	dockerPipelineObject := tektonObj.(*tektonapi.Pipeline)
+
+	// Update build-source-image param value to true
+	for i := range dockerPipelineObject.PipelineSpec().Params {
+		if dockerPipelineObject.PipelineSpec().Params[i].Name == "build-source-image" {
+			//fmt.Printf("Current build-source-image param default value: %q\n", dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal)
+			dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal = "true"
+		}
+	}
+	// Update the source-build task image reference to SOURCE_IMAGE
+	var currentSourceTaskRef string
+	for i := range dockerPipelineObject.PipelineSpec().Tasks {
+		t := &dockerPipelineObject.PipelineSpec().Tasks[i]
+		params := t.TaskRef.Params
+		var lastBundle *tektonapi.Param
+		sourceTask := false
+		for i, param := range params {
+			if param.Name == "bundle" {
+				lastBundle = &t.TaskRef.Params[i]
+			} else if param.Name == "name" && param.Value.StringVal == "source-build" {
+				sourceTask = true
+			}
+		}
+		if sourceTask {
+			currentSourceTaskRef = lastBundle.Value.StringVal
+			klog.Infof("found current task ref %s", currentSourceTaskRef)
+			lastBundle.Value = *tektonapi.NewStructuredValues(sourceImage)
+			break
+		}
+	}
+	if currentSourceTaskRef == "" {
+		klog.Errorf("failed to extract the Tekton Task from bundle: %+v", err)
+		return
+	}
+	if newPipelineYaml, err = yaml.Marshal(dockerPipelineObject); err != nil {
+		klog.Errorf("error when marshalling a new pipeline to YAML: %v", err)
+		return
+	}
+	keychain := authn.NewMultiKeychain(authn.DefaultKeychain)
+	authOption := remoteimg.WithAuthFromKeychain(keychain)
+
+	tag := fmt.Sprintf("%d-%s", time.Now().Unix(), util.GenerateRandomString(4))
+	quayOrg := utils.GetEnv(constants.DEFAULT_QUAY_ORG_ENV, constants.DefaultQuayOrg)
+	newSourceBuildPipelineImg := strings.ReplaceAll(constants.DefaultImagePushRepo, constants.DefaultQuayOrg, quayOrg)
+	var newSourceBuildPipeline, _ = name.ParseReference(fmt.Sprintf("%s:pipeline-bundle-%s", newSourceBuildPipelineImg, tag))
+
+	if err = tekton.BuildAndPushTektonBundle(newPipelineYaml, newSourceBuildPipeline, authOption); err != nil {
+		klog.Errorf("error when building/pushing a tekton pipeline bundle: %v", err)
+		return
+	}
+
+	// Update CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE env variable with the new bundle
+	// klog.Infof("SETTING ENV VAR %s to value %s\n", constants.CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE_ENV, newSourceBuildPipeline.String())
+	// err = os.Setenv(constants.CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE_ENV, newSourceBuildPipeline.String())
+	// // if err != nil {
+	// // 	klog.Errorf("error while setting env: %v\n", err)
+	// // }
```

Comment:
Are these commented lines required? The new pipeline bundle is built and pushed, but there seems no chance to update the pipeline bundle reference in the selector in `updateBuildPipelineSelector`.

Reply:
not required.. forgot to remove .. first approach I tried through `os.Setenv` which is not working.. so changed the approach and now passing the custom bundle through tekton result of `prepare-e2e-tasks` [here](https://github.com/redhat-appstudio/build-tasks-dockerfiles/pull/85/files#diff-8e21776bed38f08a61a3b9c12db92a9faed0054d07679c8b948211acec8eb352R44)

Reply:
Got it. Could you please comment this line with this info?

```go
	fmt.Printf("custom_source_image_bundle=%s\n", newSourceBuildPipeline.String())
```

Without printing the new pipeline bundle, the CI will break.

Reply:
And reading it from the e2e-tests task [here](https://github.com/redhat-appstudio/build-tasks-dockerfiles/pull/85/files#diff-0c8e9c3ca78264c1cac7adeb927c22c150aae94edbaec4ca67faf6534b91c95bR53)

Reply:
> Could you please comment this line with this info?

@tkdchen Added a comment before this line.